src/hooks/mongodb.js

```diff
@@ -1,98 +1,61 @@
-import { SpansContainer, TracerGlobals } from '../globals';
+import { hook } from '../extender';
 import * as logger from '../logger';
-import { getCurrentTransactionId } from '../spans/awsSpan';
-import { createMongoDbSpan, extendMongoDbSpan } from '../spans/mongoDbSpan';
-import { getRandomId, safeExecute } from '../utils';
+import { safeExecute } from '../utils';
 import { safeRequire } from '../utils/requireUtils';
+import { onFailedHook, onStartedHook, onSucceededHook } from './mongodb3x';
+import { afterConstructorHook, beforeConstructorHook } from './mongodb4x';
 
-const PendingRequests = (() => {
-  let pendingRequests = {};
-
-  const addPendingRequest = (requestId, spanId) => {
-    pendingRequests[requestId] = spanId;
-  };
-
-  const popPendingRequestSpanId = (requestId) => {
-    const spanId = pendingRequests[requestId];
-    delete pendingRequests[requestId];
-    return spanId;
-  };
+export const hookMongoDb = (mongoClientLibrary) => {
+  const mongoClientLibraries = mongoClientLibrary ? mongoClientLibrary : safeRequire('mongodb');
+  const mongooseClients = safeRequire('node_modules/mongoose/node_modules/mongodb');
+  const mongoClients = [mongoClientLibraries, mongooseClients].filter(Boolean);
 
-  return {
-    addPendingRequest,
-    popPendingRequestSpanId,
-  };
-})();
+  /* istanbul ignore next : there's no point in testing this */
+  if (!mongoClients) {
+    logger.debug('MongoDB clients not found');
+    return;
```

Comment:
You test setting the expected behavior. There is a point to check there is no fail. 
And you don't want to be the first one who adding `istanbul ignore` comments :)

Reply:
fair enough. and i caught unexpected behavior. thanks :)